packages/flutter/lib/src/foundation/memory_allocations.dart

```diff
@@ -0,0 +1,176 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+///
+abstract class ObjectEvent{}
+
+///
+typedef ObjectEventListener = void Function(ObjectEvent);
+
+///
+class ObjectCreated implements ObjectEvent {
+  ///
+  ObjectCreated({
+    required this.library,
+    required this.klass,
+    required this.object,
+    this.token,
+    this.details = const <Object>[],
+  });
+
+  ///
+  final String library;
+
+  ///
+  final String klass;
+
+  /// Reference to this object should not be stored in any
+  /// long living place as it will prevent garbage collection.
+  final Object object;
+
+  /// A token that uniquely identify object, for object tracking
+  /// accross events.
+  ///
+  /// If not provided, the consumers of the event may use
+  /// [identityHashCode] and handle small risk of duplicates.
+  /// The object's token should be the same accross events.
+  final Object? token;
+
+  ///
+  final List<Object> details;
+}
+
+///
+class ObjectDisposed implements ObjectEvent {
+
+  ///
+  ObjectDisposed({
+    required this.object,
+    this.details = const <Object>[],
+    this.token }
+  );
+
+  ///
+  final Object object;
+
+  ///
+  final Object? token;
+
+  ///
+  final List<Object> details;
+}
+
+
+/// The event contains some tracing information that may help with memory
+/// troubleshooting.
+///
+/// For example, it may be information about ownership transfer
+/// or state change.
+class ObjectTraced implements ObjectEvent {
+
+  ///
+  ObjectTraced(
+    this.object, {
+    this.details = const <Object>[],
+    this.token, }
+  );
+
+  ///
+  final Object object;
+
+  ///
+  final Object? token;
+
+  ///
+  final List<Object> details;
+}
+
+///
+class MemoryAllocations {
+  MemoryAllocations._();
+
+  ///
+  // Lint is ignored here, because 'late' is needed for lazy pattern.
+  // ignore: unnecessary_late
+  static late final MemoryAllocations instance = MemoryAllocations._();
```

Comment:
What's the benefit of making this lazy? 

It's not a particularly heavy duty class, and making it late will make accessing `instance` slightly more expensive.